src/controller/ReadInteraction.h

```diff
@@ -197,7 +207,16 @@ CHIP_ERROR ReportEvent(Messaging::ExchangeManager * apExchangeMgr, EndpointId en
     VerifyOrReturnError(callback != nullptr, CHIP_ERROR_NO_MEMORY);
 
     auto readClient = chip::Platform::MakeUnique<app::ReadClient>(engine, apExchangeMgr, *callback.get(), readParams.mReportType);
-    ReturnErrorOnFailure(readClient->SendRequest(readParams));
```

Comment:
Again, this is losing the "did we actually kick off the read request?" check.  And needs to null-check readClient.